.github/workflows/java-ci.yml

```
@@ -0,0 +1,40 @@
+name: Java CI
+
+on: [push]
```

```suggestion
on:
  # Trigger the workflow on push or pull request,
  # but only for the main branch
  push:
    branches:
      - master
  pull_request:
    branches:
      - master
```

This is to avoid our monthly free quota to be surpassed. Instead of running the workflow in every code push, we reserve it to pull requests and merges to master. When developing before a PR, we expect the developer to run tests locally.